cloudwatch_logs_common/src/utils/cloudwatch_facade.cpp

```diff
@@ -147,6 +149,37 @@ Aws::CloudWatchLogs::ROSCloudWatchLogsErrors CloudWatchFacade::CreateLogGroup(
   return status;
 }
 
+Aws::CloudWatchLogs::ROSCloudWatchLogsErrors CloudWatchFacade::CheckLogGroupExist(
+  const std::string & log_group)
+{
+  Aws::CloudWatchLogs::ROSCloudWatchLogsErrors status = CW_LOGS_SUCCEEDED;
+
+  Aws::CloudWatchLogs::Model::DescribeLogGroupsRequest describe_log_group_request;
+  describe_log_group_request.SetLogGroupNamePrefix(log_group.c_str());
+
+  const auto & response = this->cw_client_.DescribeLogGroups(describe_log_group_request);
+  if (!response.IsSuccess()) {
+    status = CW_LOGS_LOG_GROUP_NOT_FOUND;
+    AWS_LOGSTREAM_INFO(__func__, "Log group named: " << log_group
+            << " does not exist. Error message: "
+            << response.GetError().GetMessage() << ", with error code: "
+            << static_cast<int>(response.GetError().GetErrorType()));
+  }
+
+  if (CW_LOGS_SUCCEEDED == status) {
+    const auto & log_group_list = response.GetResult().GetLogGroups();
+    status = CW_LOGS_LOG_GROUP_NOT_FOUND;
+    for (auto it = log_group_list.begin(); it != log_group_list.end(); ++it) {
+      Aws::CloudWatchLogs::Model::LogGroup curr_log_group = *it;
+      if (curr_log_group.GetLogGroupName().c_str() == log_group) {
+        AWS_LOGSTREAM_DEBUG(__func__, "Found Log Group named: " << log_group << ".");
+        status = CW_LOGS_SUCCEEDED;
+        break;
+      }
+    }
```

Comment:
If we get out of this loop without finding the log group, because `log_group` is a proper prefix of some existing log group, then we will return CW_LOGS_LOG_GROUP_NOT_FOUND but there will be no log message. We should probably add a log message for that case too

Reply:
In fact this loop is not required because according to [the API](https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_DescribeLogGroups.html) "The results are ASCII-sorted by log group name" so if the log group exists then it will be the first result

Reply:
First comment: added logging to indicate when a log group is not found.
Second comment: you are absolutely correctly. This has been discussed in previous code reviews. The loop is just safety measure.